src/style.rs

```diff
@@ -4,16 +4,32 @@ use std::fmt::{self, Write};
 use std::mem;
 
 use console::{measure_text_width, Style};
+
 #[cfg(feature = "unicode-segmentation")]
 use unicode_segmentation::UnicodeSegmentation;
 
 use crate::format::{
     BinaryBytes, DecimalBytes, FormattedDuration, HumanBytes, HumanCount, HumanDuration,
 };
 use crate::state::ProgressState;
+use crate::StatefulFormatter;
+
+pub(crate) enum FormatterVariant {
```

Comment:
No need to have this enum, just make everything stateful. Might be nice to add a blanket impl for `impl<F: Fn(&ProgressState) -> String)> ProgressTracker for F` if possible.

Reply:
We're about to release 0.17 which IIRC is the first release to introduce this anyway. This change does not need to be additive.

Reply:
Oh is my change (tentatively) intended to be part of that release? I figured this would be delayed for something after.

Reply:
No, the 0.17.0-rc releases are from main.